distributed-lock-state-machine/distributed-lock-state-machine.md

```diff
@@ -1,15 +1,22 @@
 # Implementing a Distributed Lock State Machine
-Recently, I've been working with applications that construct Google Sheets with user-driven data.  The spreadsheet link is given to the user and the users can update the contents of the worksheet followed by clicking a hyperlink to effectively "send back" the data to the server for processing.
+A complex distributed computing problem application developers may encounter is controlled access to a resource or entity in a multi-user, concurrency environment.  How can we assuredly guard against an arbitrary number of concurrent processes from potentially mutating or even having any access to an entity simulatenously?  One possible solution is to rely on database management systems which provide concurrency safeguards via ACID transactions and row-level locking.
```

Comment:
minor suggestions:

- `concurrent environment` (?)
- `How can we guard`
- `simultaneously`